### PR TITLE
auto generate when no type

### PR DESCRIPTION
**Pull request by varHarrie**
#12

### CODE REVIEW

File: lib/create-type.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const prefix = 'auto_gen_type_'
+let index = 0
```
Comment (coderhaoxin) [inline lib/create-type.js]: why not use `uuid` ?

Comment (varHarrie) [inline lib/create-type.js]: Updated!

File: lib/create-action.js [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
 
 function noop(payload = {}) {
   return payload
 }
 
 export default function createAction(type, payloadCreator, metaCreator) {
+  if (!isString(type)) {
+    type = createType()
+    payloadCreator = type
+    metaCreator = payloadCreator
```
Comment (coderhaoxin) [inline lib/create-action.js]: this should be

```js
metaCreator = payloadCreator
payloadCreator = type
type = createType()
```

File: test/create-action.js [hunk 3]
```diff
@@ -4,6 +4,12 @@ import { equal, deepEqual } from 'assert'
 
 describe('## create-action', () => {
   describe('# basic', () => {
+    it('createAction()', () => {
+      const action = createAction()
+      equal(typeof action.type, 'string')
+      equal(action.type, action.toString())
```
Comment (coderhaoxin) [inline test/create-action.js]: need check `action works well`

File: package.json [hunk 4]
```diff
@@ -35,6 +35,7 @@
     "mocha": "2"
   },
   "dependencies": {
-    "clone": "1"
+    "clone": "1",
+    "uuid": "^3.0.1"
   }
```
Comment (coderhaoxin) [inline package.json]: prefer `"uuid": "3"`

just looks better ((~_~))!

**Comment by varHarrie** [comment]
I was so careless... :smile_cat:

File: lib/create-action.js [hunk 5]
```diff
@@ -1,11 +1,18 @@
...
+import uuid from 'uuid/v1'
 
 function noop(payload = {}) {
   return payload
 }
 
 export default function createAction(type, payloadCreator, metaCreator) {
+  if (!isString(type)) {
+    payloadCreator = type
+    metaCreator = payloadCreator
```
Comment (coderhaoxin) [inline lib/create-action.js]: this line shoule above `payloadCreator = type` 😄

**Comment by coderhaoxin** [comment]
land 515e81ab157bf6167d5968c2f282b48a770e8c60
published `v1.2.0`

**Comment by varHarrie** [comment]
@coderhaoxin 
It was not easy, this is my first PR.
Thanks for your guidance and consideration.

**Comment by coderhaoxin** [comment]
:smile:

**Pull request closed without merging**